config/manager.go

```diff
@@ -114,7 +121,11 @@ func (f *FilesystemConfigManager) SaveWith(fn SaveFunc) error {
 	}
 
 	c = fn(c)
-	return f.save(c)
+	err := f.save(c)
+	if err == nil && f.configPublisher != nil {
+		f.configPublisher.Publish(&c)
```

Comment:
Shouldn't this not be called while the mutex is locked?

Reply:
Why?

Reply:
The only concern is not to have a deadlock,  if some component will listening on this events and then tries to read the settings from here.